obs_common/license_check.py

```diff
@@ -83,29 +84,19 @@ def has_license_header(path: pathlib.Path):
     return False
 
 
-def main(args=None):
-    parser = argparse.ArgumentParser(description=DESCRIPTION)
-    parser.add_argument(
-        "-l", "--file-only", action="store_true", help="print files only"
-    )
-    parser.add_argument("--verbose", action="store_true", help="verbose output")
-    parser.add_argument("target", help="file or directory tree to check", nargs="?")
-
-    parsed = parser.parse_args(args)
-
-    if parsed.target:
-        target = pathlib.Path(parsed.target)
-        if not target.exists():
-            if not parsed.file_only:
-                print(f"Not a valid file or directory: {target}")
-            return 1
-
-        if target.is_file():
-            targets = [target]
-
-        elif target.is_dir():
-            targets = list(target.rglob("*"))
-
+@click.command(help=DESCRIPTION)
+@click.argument(
+    "targets", nargs=-1, type=click.Path(exists=True, path_type=pathlib.Path)
+)
+@click.option("-l", "--file-only", is_flag=True, help="print files only")
+@click.option("--verbose", is_flag=True, help="verbose output")
+def main(targets, file_only, verbose):
+    if targets:
+        targets = [
+            target
+            for path in targets
+            for target in (path.rglob("*") if path.is_dir() else [path])
+        ]
```

Comment:
This is pretty hard to read and reason about especially with the singular/plural variations of target.